contracts/CharityChallenge.sol

```diff
@@ -36,6 +38,9 @@ contract CharityChallenge {
     // Valid outcomes are 'YES', 'NO' and 'INVALID'
     bool public isEventFinalized;
 
+    // hasChallengeAccomplished will be set to true if we got the expected
+    // result that allow to unlock the funds.
+    // By default we expect a YES. But if unlockOnNo is true, we expect a NO.
```

Comment:
IMO, this comment `By default we expect a YES. But if unlockOnNo is true, we expect a NO.` is unclear because there's no default case or expectation for `hasChallengeAccomplished` as it depends on multiple flags. Regardless of unlockOnNo:
- If hasChallengeAccomplished is true then money is unlocked.
- If hasChallengeAccomplished is false then money is refunded.


The `unlockOnNo` affects checkAugur's logic as following:
- unlockOnNo = false:
YES => (true, false) i.e. unlock
NO or INVALID => (false, false) i.e refund
- unlockOnNo = true:
NO => (true, false) i.e. unlock
YES or INVALID => (false, false) i.e. refund

Reply:
This is how I wanted to implement it initially, but in order to not introduce one more API change (`hasChallengeAccomplished` is public and has a getter) I decided to not rename it, and adding a comment about its new meaning instead. If I could rename it, it would be `isExpectationMet`.

> Regardless of unlockOnNo:
> 
> If hasChallengeAccomplished is true then money is unlocked.
> If hasChallengeAccomplished is false then money is refunded.

This is not true anymore. From this PR, if `unlockOnNo` is true, `hasChallengeAccomplished` is negated.